### ISSUE TITLE
Attachments error if type property is missing on embedded msg files

### ISSUE THREAD

**Issue by dpjk** [TeamMsgExtractor/msg-extractor#318]
Currently, if a `.msg` file which has an `.eml` or `.msg` file as an attachment is opened using `extract_msg.openMsg()`, it will error out when it tries to get the `.eml` or `.msg` file attachment with the following:

```
../docs/email_document/email_document_mod.py:95: in convert
    self.extract_msg_docs()
../docs/email_document/email_document_mod.py:31: in extract_msg_docs
    msg = extract_msg.openMsg(self.body)
../../venv/lib/python3.8/site-packages/extract_msg/utils.py:620: in openMsg
    return Message(path, **kwargs)
../../venv/lib/python3.8/site-packages/extract_msg/message_base.py:86: in __init__
    super().__init__(path, **kwargs)
../../venv/lib/python3.8/site-packages/extract_msg/msg.py:154: in __init__
    self.attachments
../../venv/lib/python3.8/site-packages/extract_msg/msg.py:582: in attachments
    self._attachments.append(self.attachmentClass(self, attachmentDir))
../../venv/lib/python3.8/site-packages/extract_msg/attachment.py:49: in __init__
    elif (self.props['37050003'].value & 0x7) == 0x7:
_ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ 

self = {}, key = '37050003'

    def __getitem__(self, key):
>       return self.__props.__getitem__(key)
E       KeyError: '37050003'

../../venv/lib/python3.8/site-packages/extract_msg/properties.py:70: KeyError
```

Would it be possible to update the attachment gathering so that the `.eml` or `.msg` file attachments are successfully retrieved?

**Comment by TheElementalOfDestruction** [comment]
This is not related to a lack of support for embedded emails, this is due to your file missing a property that is expected to always exist on every attachment. Usually it's only checked under very specific circumstances. I need to check if the property is considered mandatory or has a listing for what to do if it's missing.

The line where it is erroring is the code trying to figure out the actual type of structure that the attachment uses, and it failing to confirm it.

**Comment by TheElementalOfDestruction** [comment]
After checking the documentation, the wording leads me to believe that the property is considered to be mandatory. It also does not specify any kind of handling for when it is missing, so I'm at a bit of a loss in where to go except to add a clarification to the error. Trying to do otherwise has the problem of being rather complicated to achieve.

Out of curiosity, were those msg files generated by outlook or some other program? If they were generated by outlook, does outlook appear to handle them properly with the ability to open the embedded files?

**Comment by dpjk** [comment]
Not certain how the `.msg` file was generated. The `.msg` file does open in outlook (using office 365 online) and appears to be fine, the email appears as an attachment that can be opened successfully within outlook. Thank you for the info!

**Comment by TheElementalOfDestruction** [comment]
Based on that, I think I can add some implementation that might try to figure it out based on that information. I'll let you know.

**Comment by TheElementalOfDestruction** [comment]
Okay, I just pushed some code to the next-release branch. Can you install from that and let me know if that fixes the issue for those msg files?

**Comment by TheElementalOfDestruction** [comment]
This should now be fixed in version 0.39.0
